### PR TITLE
Fav redesign

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2293

**Review by RZR-UA — APPROVED** [review]

File: map/src/menu/favorite/FavoriteGroupFolder.jsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    }, [markers, sortFiles]);
+
+    function getMarkersBySortFiles() {
+        let arr = [];
+        sortFiles.forEach((wpt) => {
+            const marker = markers.find((m) => m.title === wpt.name);
+            if (marker) {
+                arr.push(marker);
+            } else {
+                return [];
```
Comment (RZR-UA) [inline map/src/menu/favorite/FavoriteGroupFolder.jsx:131]: What this `return []` does?

File: map/src/manager/FavoritesManager.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
 import { apiPost } from '../util/HttpApi';
 import { quickNaNfix } from '../util/Utils';
+import TracksManager from './track/TracksManager';
+import { refreshGlobalFiles } from './track/SaveTrackManager';
+import { OBJECT_TYPE_FAVORITE } from '../context/AppContext';
 
-const FAVORITE_FILE_TYPE = 'FAVOURITES';
-const DEFAULT_GROUP_NAME = 'favorites';
+export const FAVORITE_FILE_TYPE = 'FAVOURITES';
+export const DEFAULT_FAV_GROUP_NAME = 'favorites';
```
[the remark this answers is not in the document]
Comment (alisa911) [reply map/src/manager/FavoritesManager.js:11]: favorites

**Review by RZR-UA — APPROVED** [review]

**Review by RZR-UA — APPROVED** [review]

**Review by RZR-UA — APPROVED** [review]

File: map/src/manager/FavoritesManager.js [hunk 3]
```diff
@@ -194,24 +194,25 @@ export async function saveFavoriteGroup(data, groupName, ctx) {
...
+        if (!occupied) {
+            return newName;
+        }
+        newName = name + ' - ' + i; // try with "Track - X"
+    }
+
+    if (occupied) {
+        throw new Error('FavoritesManager addTrack() too many same-tracks');
+    }
+    console.log(newName);
```
Comment (RZR-UA) [inline map/src/manager/FavoritesManager.js:213]: drop the log

**Review by RZR-UA — CHANGES_REQUESTED** [review]

**Review by RZR-UA — APPROVED** [review]

File: tests/selenium/src/tests/71-import-delete-favorite-group.mjs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    // open favorite menu
+    await clickBy(By.id('se-show-main-menu'), { optional: true });
+    await clickBy(By.id('se-show-menu-favorites'));
+
+    // delete old group when need
+    const exist = await actionCheckFileExist({ id: `se-menu-fav-${shortFavGroupName}` });
+    if (exist) {
+        await deleteFavGroup(shortFavGroupName);
+        await waitBy(By.id(`se-empty-page`));
```
Comment (RZR-UA) [inline tests/selenium/src/tests/71-import-delete-favorite-group.mjs:27]: ` to ' please

File: tests/selenium/src/util.mjs [hunk 5]
```diff
@@ -49,3 +59,15 @@ export async function deleteTrack(name) {
...
+    await actionIdleWait();
+    await waitBy(By.id(`se-folder-actions-button-${name}`));
+    await clickBy(By.id(`se-folder-actions-button-${name}`));
+    await waitBy(By.id(`se-favorite-folder-actions`));
+    await clickBy(By.id('se-favorite-folder-actions-delete'));
+    await waitBy(By.id(`se-delete-fav-group-dialog`));
+    await clickBy(By.id('se-delete-fav-group-submit'));
+    await waitBy(By.id(`se-actions-${name}`), { hidden: true });
+    await actionIdleWait();
+}
```
Comment (RZR-UA) [inline tests/selenium/src/util.mjs:73]: Please change ` to ' everywhere you don't need ${}

**Review by RZR-UA — CHANGES_REQUESTED** [review]

File: tests/selenium/src/tests/73-edit-favorite.mjs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            await input.sendKeys(suffix);
+            return true;
+        },
+        { tag: 'edit-address-fav-item' }
+    );
+
+    await clickBy(By.id('se-edit-fav-item-submit'));
+    await waitBy(By.id(`se-actions-${wptName}`), { hidden: true });
+
+    await matchTextBy(By.id('se-fav-item-address'), `${suffix}`);
```
Comment (RZR-UA) [inline tests/selenium/src/tests/73-edit-favorite.mjs:53]: `${suffix}` -> suffix

**Review by RZR-UA — CHANGES_REQUESTED** [review]

File: map/src/dialogs/favorites/DeleteFavoriteDialog.jsx [hunk 7]
```diff
@@ -93,13 +93,15 @@ export default function DeleteFavoriteDialog({ dialogOpen, setDialogOpen, wpt =
...
     return (
-        <Dialog open={true} onClose={toggleShowDialog} onClick={(e) => e.stopPropagation()}>
+        <Dialog id="se-delete-fav-dialog" open={true} onClose={toggleShowDialog} onClick={(e) => e.stopPropagation()}>
             <DialogTitle>{getTitleDialog()}</DialogTitle>
             <DialogContent>
                 <DialogContentText>{getQuestionDialog()}</DialogContentText>
             </DialogContent>
             <DialogActions>
-                <Button onClick={() => deleteWpt()}>Delete</Button>
+                <Button id="se-delete-fav-dialog-submit" onClick={() => deleteWpt()}>
```
Comment (RZR-UA) [inline map/src/dialogs/favorites/DeleteFavoriteDialog.jsx:102]: When you don't need any params for event handler, please contract the call:

`onClick={() => deleteWpt()}` -> onClick={deleteWpt}

The same as you've already done at line 96 for onClose.

File: tests/selenium/src/tests/74-delete-favorite.mjs [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+    // delete favorite item
+    await clickBy(By.id('se-delete-fav-dialog-submit'));
+    await waitBy(By.id(`se-actions-${wptName}`), { hidden: true });
+    await waitBy(By.id(`se-fav-item-${wptName}`), { hidden: true });
+
+    await clickBy(By.id('se-back-folder-button'));
+    await waitBy(By.id(`se-menu-fav-${shortFavGroupName}`));
+
+    await deleteFavGroup(`${shortFavGroupName}`);
```
Comment (RZR-UA) [inline tests/selenium/src/tests/74-delete-favorite.mjs:49]: -> shortFavGroupName

File: tests/selenium/src/tests/74-delete-favorite.mjs [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    await waitBy(By.id(`se-empty-page`));
```
Comment (RZR-UA) [inline tests/selenium/src/tests/74-delete-favorite.mjs:50]: Please stop using `` instead of ''

**Review by RZR-UA — CHANGES_REQUESTED** [review]

**Pull request merged**